tests/test_ensemble_tools.py

```diff
@@ -0,0 +1,163 @@
+    for i in range(n_dirs):
+        cond_file = test_sweep_dir / "SWEEP" / f"asd{i}" / "conditions.xml"
+        assert cond_file.is_file()
+        para_value = scan_range[0] + (i / (n_dirs - 1)) * (
+            scan_range[1] - scan_range[0]
+        )
+        assert _check_parameter_in_conditions(cond_file, parameter_to_scan, para_value)
+    # Should raise an exception when trying to edit the mesh file
+    edit_file = "mesh.xml"
+    with pytest.raises(ValueError, match=r".* a CONDITIONS node.*"):
+        create_dir_tree(
+            sweep_path=test_sweep_dir,
+            n_dirs=n_dirs,
+            destructive=destructive,
+            copy_dir=copy_dir,
+            edit_file=edit_file,
+            parameter_to_scan=parameter_to_scan,
+            scan_range=scan_range,
+            outdir_prefix=outdir_prefix,
+        )
+
+
+def test_create_dict_sweep(tmpdir: local) -> None:
+    """Test the create_dict_sweep method of ensemble_tools."""
+    sweep_path = Path(tmpdir) / "test"
+    n_divs = 5
+    destructive = True
+    copy_dir = Path("config_files") / "two_stream"
+    edit_file = "conditions.xml"
+    parameter_dict = {
+        "particle_initial_velocity": (0.1, 2.5),
+        "particle_charge_density": (102.0, 108.0),
+    }
+    create_dict_sweep(
+        sweep_path=sweep_path,
+        n_divs=n_divs,
+        destructive=destructive,
+        copy_dir=copy_dir,
+        edit_file=edit_file,
+        parameter_dict=parameter_dict,
+    )
+    n_total_directories = n_divs ** len(parameter_dict)
+    # Check we make the corect number of directories
+    assert len(list((sweep_path / "SWEEP").iterdir())) == n_total_directories
+
+    # Loop through the directories and check the conditions file
+    for indices in itertools.product(*(range(n_divs),) * len(parameter_dict)):
+        directory_name = "-".join(
+            f"{k}_{i}" for k, i in zip(parameter_dict, indices, strict=True)
+        )
+        this_dir = sweep_path / "SWEEP" / directory_name
+
+        # Check the directory exists
+        assert this_dir.is_dir()
+
+        # Check the conditions and mesh files exist
+        assert (this_dir / "conditions.xml").is_file()
+        assert (this_dir / "mesh.xml").is_file()
+
+        # Check that the parameters have been edited correctly
+        for i in range(len(parameter_dict)):
+            parameter = list(parameter_dict.keys())[i]
+            scan_range = parameter_dict[parameter]
+            para_value = scan_range[0] + (indices[i] / (n_divs - 1)) * (
+                scan_range[1] - scan_range[0]
+            )
+            assert _check_parameter_in_conditions(
+                this_dir / "conditions.xml", parameter, para_value
+            )
+            with pytest.raises(ValueError, match=r".*CONDITIONS.*"):
+                # The mesh file should be not edited
+                assert _check_parameter_in_conditions(
+                    this_dir / "mesh.xml", parameter, para_value
+                )
```

Comment:
Not sure if it makes sense to have an `assert` within the `pytest.raises` here, and again I would probably separate out checks for expected exception behaviour  in to a separate test function.